pkg/standalone-utils/contracts/LPOracleFactoryBase.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { ILPOracleFactoryBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleFactoryBase.sol";
+import { ILPOracleBase } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/ILPOracleBase.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+
+abstract contract LPOracleFactoryBase is ILPOracleFactoryBase, SingletonAuthentication {
+    uint256 internal _oracleVersion;
+    mapping(IBasePool => ILPOracleBase) internal _oracles;
+    mapping(ILPOracleBase => bool) internal _isOracleFromFactory;
+
+    constructor(IVault vault, uint256 oracleVersion) SingletonAuthentication(vault) {
```

Comment:
Is the only reason for `SingletonAuthentication` to support `getVault`? (vs. granting the ability to have permissioned functions in derived oracles - doesn't seem there are any in weighted or stable, so oracles that need that could include it at a lower level).

If so, consider storing it here: `IVault private immutable _vault`, define a public `getVault`, and just use `_vault` internally.

Reply:
having `disable` could be a good idea (same as pool factories: do not allow creating more oracles if disabled; existing oracles keep working as usual). Let's add it too.